docarray/index/backends/epsilla.py

```diff
@@ -0,0 +1,510 @@
+                    f"Error code: {status_code}. Error message: {response}."
+                )
+
+            # Epsilla cloud requires table to be created in the web UI before inserting data
+            # It does not support creating tables from Python client yet.
+
+    def _validate_column_info(self):
+        vector_columns = []
+        for info in self._column_infos.values():
+            for type in [list, np.ndarray, AbstractTensor]:
+                if safe_issubclass(info.docarray_type, type) and info.config.get(
+                    'is_embedding', False
+                ):
+                    # check that dimension is present
+                    if info.n_dim is None and info.config.get('dim', None) is None:
+                        raise ValueError("The dimension information is missing")
+
+                    vector_columns.append(info.docarray_type)
+                    break
+
+        if len(vector_columns) == 0:
+            raise ValueError(
+                "Unable to find any vector columns. Please make sure that at least one "
+                "column is of a vector type with the is_embedding=True attribute specified."
+            )
+        elif len(vector_columns) > 1:
+            raise ValueError("Specifying multiple vector fields is not supported.")
+
+    def _create_table_self_hosted(self):
+        """Use _column_infos to create a table in the database."""
+        table_fields = []
+
+        primaryKeys = []
+        for column_name, column_info in self._column_infos.items():
+            if column_info.docarray_type == ID:
+                primaryKeys.append(column_name)
+
+        # when there is a nested schema, we may have multiple "ID" fields. We use the presence of "__"
+        # to determine if the field is nested or not
+        if len(primaryKeys) > 1:
+            sorted_pkeys = sorted(primaryKeys, key=lambda x: x.count("__"))
+            primaryKeys = sorted_pkeys[:1]
+
+        for column_name, column_info in self._column_infos.items():
+            dim = (
+                column_info.n_dim
+                if column_info.n_dim is not None
+                else column_info.config.get('dim', None)
+            )
+            if dim is None:
+                table_fields.append(
+                    {
+                        'name': column_name,
+                        'dataType': column_info.db_type,
+                        'primaryKey': column_name in primaryKeys,
+                    }
+                )
+            else:
+                table_fields.append(
+                    {
+                        'name': column_name,
+                        'dataType': column_info.db_type,
+                        'dimensions': dim,
+                    }
+                )
+
+        status_code, response = self._db.create_table(
+            table_name=self._table_name,
+            table_fields=table_fields,
+        )
+        if status_code != 200:
```

Comment:
is there a constant feom the library that could allow us to avoid using magic numbers?